src/evm_context.rs

```diff
@@ -366,11 +366,8 @@ pub fn staticcall(
 /// Model an unknown write to a section of memory.
 /// Currently makes all of memory unknown if the length is nonzero.
 #[must_use]
-fn unknown_memory_write(location: &MemoryRange, ssa: &mut SSATracker) -> Vec<SMTStatement> {
-    ssa.havoc_unless(
-        &context().memory.identifier,
-        smt::eq_zero(location.length.clone()),
-    )
+fn unknown_memory_write(location: MemoryRange, ssa: &mut SSATracker) -> Vec<SMTStatement> {
+    ssa.havoc_unless(&context().memory.identifier, smt::eq_zero(location.length))
```

Comment:
This is weird - does `location.length` move? Why don't we need `std::mem::take`? Or does it move if the struct is by value and it does not move if it is by mutable ref?

Reply:
I think it moves. I left the `clone` at first, but clippy said I should remove it so...